src/engine/interop/specification.md

```diff
@@ -0,0 +1,148 @@
+# Engine API. Interop edition
+
+This document specifies a subset of the Engine API methods that are required to be implemented for the Merge interop.
+
+*Note*: Sync API design is yet a draft and considered optional for the interop.
+
+## Underlying protocol
+
+This specification is based on [Ethereum JSON-RPC API](https://eth.wiki/json-rpc/API) and inherits the following properties of this standard:
+* Supported communication protocols (HTTP and WebSocket)
+* Message format and encoding notation
+* [Error codes improvement proposal](https://eth.wiki/json-rpc/json-rpc-error-codes-improvement-proposal)
+
+Client software **MUST** expose Engine API at a port independent from JSON-RPC API. The default port for the Engine API is 8550 for HTTP and 8551 for WebSocket.
+
+## Error codes
+
+The list of error codes introduced by this specification can be found below.
+
+| Code | Possible Return message | Description |
+| - | - | - |
+| 4 | Missing resource | Should be used when one of the call parameters references a missing resource and the call can't be processed, e.g. some action over unknown block is requested |
+
+## Structures
+
+Encoding values of `DATA` and `QUANTITY` types **MUST** follow [HEX value encoding](https://eth.wiki/json-rpc/API#hex-value-encoding) section of Ethereum JSON-RPC API.
+
+*Note:* Big-endian byte order **MUST** be used to encode values of `QUANTITY` type.
```

Comment:
Does data also need to be specified as big endian?

Reply:
`DATA` is a sequence of bytes without a notion of endianness